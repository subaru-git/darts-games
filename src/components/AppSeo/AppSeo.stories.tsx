import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import AppSeo from './AppSeo';

export default {
  title: 'Components/AppSeo',
  component: AppSeo,
} as ComponentMeta<typeof AppSeo>;

const Template: ComponentStory<typeof AppSeo> = (args) => <AppSeo {...args} />;

export const Default = Template.bind({});
Default.args = {
  title: 'AppSep Stories',
  description: 'AppSep Stories.',
  canonical: 'https://darts.homes',
};
