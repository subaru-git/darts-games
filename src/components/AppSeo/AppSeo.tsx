import React, { FC } from 'react';
import { NextSeo, NextSeoProps } from 'next-seo';

type AppSetProps = {
  title: string;
  description: string;
  canonical: string;
};

const AppSeo: FC<NextSeoProps & AppSetProps> = (props) => {
  const { title, description, canonical } = props;
  return <NextSeo {...props} title={title} description={description} canonical={canonical} />;
};

export default AppSeo;
