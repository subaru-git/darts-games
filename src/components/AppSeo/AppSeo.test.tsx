import { render } from '@testing-library/react';
import AppSeo from './AppSeo';

jest.mock('next/router', () => ({ useRouter: jest.fn().mockReturnValue({ locale: 'en' }) }));
jest.mock('next/head', () => {
  return {
    __esModule: true,
    default: ({ children }: { children: Array<React.ReactElement> }) => {
      return <>{children}</>;
    },
  };
});

test('should rendering', () => {
  const { container } = render(<AppSeo title='title' description='description' canonical='canonical' />, { container: document.head });
  expect(document.title).toBe('title');
  expect(container).toMatchSnapshot();
});
