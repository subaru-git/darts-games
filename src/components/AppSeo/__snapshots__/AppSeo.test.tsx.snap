// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should rendering 1`] = `
<head>
  <title>
    title
  </title>
  <meta
    content="index,follow"
    name="robots"
  />
  <meta
    content="description"
    name="description"
  />
  <meta
    content="title"
    property="og:title"
  />
  <meta
    content="description"
    property="og:description"
  />
  <link
    href="canonical"
    rel="canonical"
  />
</head>
`;
